test/unit/product/product-component.js

```diff
@@ -761,52 +761,50 @@ describe('Product Component class', () => {
+          assert.isTrue(product.shouldUpdateImage);
         });
 
-        describe('if image and cached image are different', () => {
-          beforeEach(async () => {
-            product.config.product.width = '100px';
-            await product.init(testProductCopy);
-          });
-
-          it('returns true', () => {
-            product.cachedImage = 'bar.jpg';
-            assert.ok(product.shouldUpdateImage);
-          });
+        it('returns true if image and cached image are different', async () => {
+          product.config.product.width = '100px';
+          await product.init(testProductCopy);
+          product.cachedImage = 'bar.jpg';
+          assert.isTrue(product.shouldUpdateImage);
         });
 
-        describe('if image and cached image are same', () => {
-          beforeEach(async () => {
-            product.config.product.width = '240px';
-            await product.init(testProductCopy);
-          });
-
-          it('returns true', () => {
-            product.cachedImage = `${rootImageURI}image-one_240x360.jpg`;
-            assert.notOk(product.shouldUpdateImage);
-          });
+        it('returns false if image and cached image are the same', async () => {
+          product.config.product.width = '240px';
+          await product.init(testProductCopy);
+          product.cachedImage = `${rootImageURI}image-one_240x360.jpg`;
+          assert.isFalse(product.shouldUpdateImage);
         });
       });
 
+
       describe('currentImage', () => {
-        describe('if variant exists', () => {
-          it('returns selected image', async () => {
-            await product.init(testProductCopy);
-            assert.equal(product.currentImage.src, `${rootImageURI}image-one_280x420.jpg`);
+        it('sets cached image to product.image if image should update', async () => {
+          product = Object.defineProperty(product, 'shouldUpateImage', {
+            value: true,
           });
+          await product.init(testProductCopy);
+          assert.deepEqual(product.cachedImage, product.image);
         });
 
-        describe('if variant does not exist', () => {
-          it('returns cached image', async () => {
-            await product.init(testProductCopy);
-            product.selectedVariant = {};
-            assert.equal(product.currentImage.src, `${rootImageURI}image-one_280x420.jpg`);
+        it('does not set cached image to product.image if image should not update', async () => {
+          product = Object.defineProperty(product, 'shouldUpateImage', {
+            value: true,
           });
+          product.cachedImage = {
+            src: 'not-image-src',
+          };
+          await product.init(testProductCopy);
+          assert.notEqual(product.cachedImage, product.image);
+        });
+
+        it('returns cached image', async () => {
+          await product.init(testProductCopy);
+          assert.equal(product.currentImage, product.cachedImage);
```

Comment:
Can you make this test more explicit on why the `product.currentImage` is equal to `product.cachedImage`?